src/backend/distributed/metadata/dependency.c

```diff
@@ -741,6 +741,124 @@ SupportedDependencyByCitus(const ObjectAddress *address)
 }
 
 
+/*
+ * EnsureRelationDependenciesCanBeDistributed ensures all dependencies of the relation
+ * can be distributed.
+ */
+void
+EnsureRelationDependenciesCanBeDistributed(ObjectAddress *relationAddress)
+{
+	ObjectAddress *undistributableDependency =
+		GetUndistributableDependency(relationAddress);
+
+	if (undistributableDependency != NULL)
+	{
+		char *tableName = get_rel_name(relationAddress->objectId);
+
+		if (SupportedDependencyByCitus(undistributableDependency))
+		{
+			/*
+			 * Citus can't distribute some relations as dependency, although those
+			 * types as supported by Citus. So we can use get_rel_name directly
+			 */
+			char *dependentRelationName = get_rel_name(
+				undistributableDependency->objectId);
+
+			ereport(ERROR, (errmsg("Relation \"%s\" has dependency on non-distributed "
```

Comment:
how are we sure that this is a "relation"? (e.g., should we change the error message to object)

Reply:
I guess we do this in other places as well, because we know that GetUndistributableDependency && SupportedDependencyByCitus can only be relation.

That seems could easily be broken in the future. Can we expand the message to maybe rely on `getObjectTypeDescription` as well?

Reply:
I've kept the relation message by adding an explicit check as I don't want to show confusing message for a known case.